src/core/services/LabService.ts

```diff
@@ -141,15 +142,18 @@ export class LabServiceImpl implements LabService {
     let yvBoostLab: Lab | undefined;
     try {
       const yvBoostContract = getContract(YVBOOST, yvBoostAbi, provider);
-      const [totalAssets, pricePerShare, depositLimit, emergencyShutdown] = await Promise.all([
-        yvBoostContract.totalAssets(),
-        yvBoostContract.pricePerShare(),
-        yvBoostContract.depositLimit(),
-        yvBoostContract.emergencyShutdown(),
-      ]);
+      const totalAssets = await yvBoostContract.totalAssets();
```

Comment:
I think we can also get this value from `yearn.vaults.getDynamic` result. Try `yvBoostVaultDynamic.underlyingTokenBalance.amount` if its the same as `totalAssets`. If that is correct, then also use the `amountUsdc` value from response instead of us continuing to calculated manually.

Reply:
You're right, they are the same:

```typescript
totalAssets.toString()
// "12970596589987347531198138"

yvBoostVaultDynamic.underlyingTokenBalance.amount.toString()
// "12970596589987347531198138"
```

Refactored with https://github.com/yearn/yearn-finance-v3/pull/558/commits/0dc62b3fcc4ee2f9272fc733645c17f74d8d11d8 to used getDynamic

Reply:
Yes, that `pricesResponse` comes from coingecko and `yvBoostVaultDynamic` from oracle contract, so should be more real, unless there is low liquidity. If prices are similar, lets go with the later, but we need to make sure price is consistent here too https://github.com/yearn/yearn-finance-v3/blob/develop/src/core/services/TokenService.ts#L116 if we decide to change it.

Reply:
Refactored with https://github.com/yearn/yearn-finance-v3/pull/558/commits/0f3e5cfe76f529a412a520457113b908b03e68cf to use the balances from the `underlyingTokenBalance` from the getDynamic vaults.